### DISCUSSION TITLE
Trouble with unicode

### DISCUSSION THREAD

**Discussion by omerfkati** [py-pdf/fpdf2#366] [Q&A] [answered]
Hi fpdf2 team,

I can't seem to get the unicode to work properly. I am generating a dutch pdf and there is text with the letters é and ë. The pdf creator does not seem to understand them. 

I am using the Roboto font:
pdf.add_font("Roboto", "", r"./fonts/Roboto-Regular.ttf", uni=True)

However the same problem occurs with the Dejavu font.

I also see that the uni=True part is deprecated, but no alternative is given. How should we implement unicode then? The tutorial seems to be dated as well since it mentions uni=True. I only need those two letters really so any help would be greatly appreciated.

**Comment by Lucas-C** [comment, marked as answer]
> I can't seem to get the unicode to work properly. I am generating a dutch pdf and there is text with the letters é and ë. The pdf creator does not seem to understand them.

Could you please provide some [minimal but fully functional Python code reproducing your issue](https://stackoverflow.com/help/minimal-reproducible-example),
in order for use to be able to help you?

> I also see that the uni=True part is deprecated, but no alternative is given. How should we implement unicode then?

Quoting the [ChangeLog](https://github.com/PyFPDF/fpdf2/blob/master/CHANGELOG.md#251---2022-03-07):

`uni=True` can now be removed from all calls to `FPDF.add_font()`.
If the value of the `fname` argument passed to `add_font()` ends with `.ttf`, it is considered a TrueType font.

> The tutorial seems to be dated as well since it mentions uni=True. I only need those two letters really so any help would be greatly appreciated.

Sorry but where have you spotted any `uni=True`?
I haven't found any in this tutorial page: https://pyfpdf.github.io/fpdf2/Tutorial.html

**Comment by omerfkati** [comment]
I found the problem. I was not reading the file with the utf-8 encoding. Everything works now thanks for the help. 

I was referring to this page:
https://pyfpdf.github.io/fpdf2/fpdf/fpdf.html#fpdf.fpdf.FPDF.add_font

But I understand now that there is no need for the uni=True option anymore. Sorry for the trouble.

**Reply by Lucas-C** [reply]
Great! Glad that it works fine for you now 😊
